consensus/propagation/commitment.go

```diff
@@ -170,7 +170,7 @@ func (blockProp *Reactor) handleCompactBlock(cb *proptypes.CompactBlock, peer p2
 		case blockProp.proposalChan <- cb.Proposal:
 		}
 		// check if we have any transactions that are in the compact block
-		go blockProp.recoverPartsFromMempool(cb)
+		blockProp.recoverPartsFromMempool(cb)
```

Comment:
this is beneficial because we don't want to act on haves before we know what we have locally and what we should request

Reply:
I think the goroutine was added as a premature optimization. Even if the mempool is very congested, retrieving transactions from the mempool should be fast enough and if we have any local parts, it's better to retrieve them rather than request them. If this process is slow, I guess the right is to speed it up instead of making it run in parallel.